vendors/espressif/boards/esp32/ports/ble/nimble/iot_ble_hal_gap.c

```diff
@@ -624,32 +624,65 @@ BTStatus_t prvBTSetAdvData( uint8_t ucAdapterIf,
         fields.svc_data_uuid128_len = usServiceDataLen;
     }
 
-    if( pxServiceUuid != NULL )
+    fields.num_uuids16=0;
+    fields.num_uuids32=0;
+    fields.num_uuids128=0;
+    for (size_t i = 0; i < xNbServices; i++)
     {
-        if( pxServiceUuid->ucType == eBTuuidType16 )
+      if( &(pxServiceUuid[i]) != NULL )
```

Comment:
This check is not needed as  &(pxServiceUuid[i]) can never be NULL.

It makes sense to check for `if( pxServiceUuid != NULL )` before iterating through the array here `for (size_t i = 0; i < xNbServices; i++)`

Reply:
`pxServiceUuid[i]` can never be NULL only if we assume the calling code has no bugs.
I think the check is needed as a form of defensive coding, because the caller might have passed a wrong `xNbServices` (i.e. one that is inconsistent with `pxServiceUuid`).
e.g. if the caller passed no services in `pxServiceUuid`, but set `xNbServices` to `1`, removing this check would result in an out-of-bounds memory access.

Reply:
@matteoscordino 
Array out of bounds cannot be verified using the method you mentioned. Example if array `pxServiceUuid` points to address say `0x1000` and each element size is `K` then  `&pxServiceUuid[N+1]` will be pointing to address `(0x1000 + ( N+1) * K` and this will never be `null(0)`. This would work if `pxServiceUuid is NULL` and user passes `xNbServices > 0` but a better check would be `pxServiceUuid  != NULL` in terms of readability.

Reply:
As you prefer, I will make the change.